ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/WithdrawalsValidatorProvider.java

```diff
@@ -15,17 +15,42 @@
 
 package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine;
 
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.BlockHeaderBuilder;
 import org.hyperledger.besu.ethereum.mainnet.ProtocolSpec;
 import org.hyperledger.besu.ethereum.mainnet.TimestampSchedule;
 import org.hyperledger.besu.ethereum.mainnet.WithdrawalsValidator;
 
+import java.util.Optional;
+
 public class WithdrawalsValidatorProvider {
 
   static WithdrawalsValidator getWithdrawalsValidator(
-      final TimestampSchedule timestampSchedule, final long newPayloadTimestamp) {
+      final TimestampSchedule timestampSchedule,
+      final long blockTimestamp,
+      final long blockNumber) {
+
+    final BlockHeader blockHeader =
+        BlockHeaderBuilder.createDefault()
+            .timestamp(blockTimestamp)
+            .number(blockNumber)
+            .buildBlockHeader();
+    return Optional.ofNullable(timestampSchedule.getByBlockHeader(blockHeader))
+        .map(ProtocolSpec::getWithdrawalsValidator)
+        // TODO Withdrawals this is a quirk of the fact timestampSchedule doesn't fallback to the
+        // previous fork. This might be resolved when
+        // https://github.com/hyperledger/besu/issues/4789 is played
+        // and if we can combine protocolSchedule and timestampSchedule.
+        .orElseGet(WithdrawalsValidator.ProhibitedWithdrawals::new);
```

Comment:
Can the duplicated code between this function and the other getWithdrawalsValidator function be moved into a common function

Reply:
As eluded to in the TODO comment, this fallback will disappear when we unify the schedules in the next PR